backend/app/src/main/java/com/moh/phlat/backend/repository/SourceDataFilterSpecificationImpl.java

```diff
@@ -0,0 +1,35 @@
+package com.moh.phlat.backend.repository;
+
+import java.util.List;
+
+import org.springframework.data.jpa.domain.Specification;
+import com.moh.phlat.backend.model.ProcessData;
+import com.moh.phlat.backend.model.SourceData;
+
+import jakarta.persistence.criteria.JoinType;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor
+public class SourceDataFilterSpecificationImpl implements SourceDataFilterSpecification {
+	
+	@Override
+	public Specification<SourceData> buildSpecificationWhereEqual(String columnKey, String value){
+		return Specification.where((root, query, builder) -> builder.equal(root.get(columnKey), value));
```

Comment:
value will require blank, null check

Reply:
Added the hastext and throws an exception if the value is missing.